######################
#Iodamoeba clustering
######################
#first pull iodamoeba otu ids that made it into the placement tree, filter from representative set
cd /parfreylab/mann/primateEuk
mkdir iodamoeba_heatmap
grep "denovo" eukref_trees/archamoeba_placement/htes_aligned_iodamoeba/iodamoeba.denovo_aligned.plus_curated_pfiltered.fasta | sed 's/>//' > iodamoeba_heatmap/ids
cd iodamoeba_heatmap
filter_fasta.py -f /parfreylab/mann/primateEuk/data/v4/swarm/rep_set.fa -s ids -o iodamoeba.fa

#now add in reference iodamoeba sequences and align
grep "Ioda" ../eukref_trees/archamoeba_reference/reference.amoeba.fa | awk '{print $1}' | sed 's/>//' > refids
filter_fasta.py -f /parfreylab/mann/primateEuk/eukref_trees/archamoeba_reference/reference.amoeba.fa -s refids -o refs.fa
cat iodamoeba.fa refs.fa > full.fa
mafft --auto full.fa > align.fa

#clean up alignment
vagrant up
vagrant scp Downloads/align.fa :/home/vagrant/iodamoeba
vagrant ssh
trimal -in align.fa -out align.trimal.fa -gt 0.99 -st 0.001
scp -P 2222 vagrant@127.0.0.1:/home/vagrant/iodamoeba/align* .
sed -i 's/ .*//' align.clean.fa
#open in aliviewer, remove sequences that are too short for proper alignment, now everything down to 171 bp
#dereplicate so that reads that are now identical are removed

#build tree
scp align.trimal.fa align.clean.fa mann@zoology.ubc.ca:/parfreylab/mann/primateEuk/iodamoeba_heatmap
raxmlHPC-PTHREADS-SSE3 -T 4 -m GTRCAT -c 25 -e 0.001 -p 31415 -f a -N 100 -x 02938 -n tre -s align.clean.fa

#build tree with reduced (duplicate removed) file generated by raxml
raxmlHPC-PTHREADS-SSE3 -T 4 -m GTRCAT -c 25 -e 0.001 -p 31414 -f a -N 100 -x 02938 -n dedup.tre -s align.clean.fa.reduced

#fix reduced file
scp mann@zoology.ubc.ca:/parfreylab/mann/primateEuk/iodamoeba_heatmap/*reduced* .
sed -i 's/^/>/' align.clean.fa.reduced
sed -i 's/ /\n/' align.clean.fa.reduced

#want to get a distance matrix in r
library(ape)
library(RColorBrewer)
library(gplots)
align <- read.dna("align.clean.fa.reduced", format="fasta")
#Kimura 1980 distance calculation, considers transitions/transversion as diff probability
mat <- dist.dna(align, as.matrix=TRUE, model="K80")
tree <- read.tree("RAxML_bipartitions.dedup.ordered.tre")
tree <- root(tree, outgroup="JN635740.1")
neworder <- tree$tip.label
mat <- mat[neworder,neworder]
pal <- colorRampPalette(rev(brewer.pal(10, 'YlOrRd')))
pdf("iodamoeba_dist_heatmap.pdf")
heatmap.2(mat, symm=T, col=pal(9), key=TRUE, density.info="none", trace="none")
dev.off()

#got denovo ids from tree (using archaeopteryx)
#now pull these together into a dataframe for permanova analysis -- are the different clusters significantly different from one another?
sed -i 's/\t1/\tcluster1/' cluster1.ids
sed -i 's/\t1/\tcluster2/' cluster2.ids
cat cluster1.ids cluster2.ids > for.permanova.txt
#add in headers, load into r with distance matrix
library(ape)
library(RColorBrewer)
library(gplots)
align <- read.dna("align.clean.fa.reduced", format="fasta")
#Kimura 1980 distance calculation, considers transitions/transversion as diff probability
mat <- dist.dna(align, as.matrix=TRUE, model="K80")
tree <- read.tree("RAxML_bipartitions.dedup.ordered.tre")
tree <- root(tree, outgroup="JN635740.1")
